utils/session.c

```diff
@@ -655,6 +655,57 @@ struct sym * task_find_sym_addr(struct uftrace_session_link *sessions,
 	return sym;
 }
 
+/**
+ * task_find_loc_addr - find a debug location that matches to @addr
+ * @sessions: session link to manage sessions and tasks
+ * @task: handle for functions in a task
+ * @time: timestamp of the @addr
+ * @addr: instruction address
+ *
+ * This function returns a debug location of symbol
+ * that looked up in symbol table in current session
+ */
+struct debug_location * task_find_loc_addr(struct uftrace_session_link *sessions,
+				struct uftrace_task_reader *task,
+				uint64_t time, uint64_t addr)
+{
+	struct uftrace_session *sess;
+	struct sym *sym = NULL;
+	struct uftrace_mmap *map;
+
+	sess = find_task_session(sessions, task->t, time);
+
+	if (sess == NULL) {
+		struct uftrace_session *fsess = sessions->first;
+
+		if (is_kernel_address(&fsess->symtabs, addr))
+			sess = fsess;
+		else
+			return NULL;
+	}
+
+	sym = find_symtabs(&sess->symtabs, addr);
+	if (sym == NULL)
+		sym = session_find_dlsym(sess, time, addr);
```

Comment:
Can sym actually be null?
if sym can be null, I need to check it in next line before accessing it.

Reply:
If addr is not found in the symbol table, it can be NULL.

Reply:
Thank you for your information.
It checked in line 691 to return NULL when it's NULL.